src/status_im/hardwallet/core.cljs

```diff
@@ -82,16 +260,146 @@
 (defn- proceed-to-pin-confirmation [fx]
   (assoc-in fx [:db :hardwallet :pin :enter-step] :confirmation))
 
-(defn- pin-match [fx]
-  (assoc-in fx [:db :hardwallet :pin :status] :validating))
+(fx/defn pin-match
+  [{:keys [db] :as fx}]
+  (let [pairing (get-in db [:hardwallet :secrets :pairing])
+        new-pin (apply str (get-in db [:hardwallet :pin :original]))
```

Comment:
Those apply str could be in a separate fn with some doc :)

Reply:
@jeluard moved to separate fn